divio/djangocms-versioning-filer
# Correct the mistake of joining with file.pk

What happens here is that it uses the id from grouper and it tries
to use that id to join with the file table id.

The problem is that grouper_id is not the same as the file pk so
you'll end up with a file you weren't looking for.

So instead we join on grouper id that exists on the file table.

-- 

An unpublished asset may not exist on target website and
end up as a dead link. To prevent that we check that the
asset is published as part of adding the asset to a field.

**How to test manually**

Upload an image, the uploaded image will now be in draft. Create a new page, try to add that image in an image panel. You'll get a validation error saying that the image is not published. 

Make sure that the image is published. Try add the image again. When saving the panel, the correct image has been added.
